clients/ui/bff/internal/api/middleware.go

```diff
@@ -211,20 +213,20 @@ func (app *App) PerformSARonSpecificService(next func(http.ResponseWriter, *http
 			return
 		}
 
-		modelRegistryID := ps.ByName(ModelRegistryId)
-		if !ok || modelRegistryID == "" {
+		serviceName := ps.ByName(ModelRegistryId)
+		if !ok || serviceName == "" {
 			app.badRequestResponse(w, r, fmt.Errorf("missing namespace in context"))
 			return
 		}
 
-		var userGroups []string
-		if groups, ok := r.Context().Value(constants.KubeflowUserGroupsKey).([]string); ok {
-			userGroups = groups
-		} else {
-			userGroups = []string{}
+		client, err := app.kubernetesClientFactory.GetClient(ctx)
+		if err != nil {
+			app.serverErrorResponse(w, r, fmt.Errorf("failed to get Kubernetes client: %w", err))
+			return
 		}
 
-		allowed, err := app.kubernetesClient.PerformSARonSpecificService(user, userGroups, namespace, modelRegistryID)
+		allowed, err := client.CanAccessServiceInNamespace(r.Context(), identity, namespace, serviceName)
+
 		if err != nil {
 			app.forbiddenResponse(w, r, "failed to perform SAR: %v")
```

Comment:
It's good security practice not to return any additional information to the user for an authentication response as you can leak clues to the internal architecture of the system, that could lead to CWEs like a response discrepancy.

It's probably a good idea to alter the forbiddenResponse function to not write the error message to the http response and just log it, whilst returning simply 403 forbidden.

Reply:
Thank you for the review!